lib/takeaway.rb

```diff
@@ -0,0 +1,60 @@
+require 'messenger'
+
+class Takeaway
+  attr_accessor :menu, :items, :balance
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @items = Hash.new(0)
+    @balance = 0
+  end
+
+  def read_menu
+    @menu.dishes
+  end
+
+  def add_order(item, quantity = 1)
+    @items[item] += quantity
+    @balance += @menu.dishes[item] * quantity
```

Comment:
Is it worth keeping an updated balance before the user tries to check out? As it will need to be continuously updated if items are added / removed, vs. calculating the total at checkout once.